examples/platforms/simulation/system.c

```diff
@@ -63,56 +64,101 @@ static void handleSignal(int aSignal)
     gTerminate = true;
 }
 
+/**
+ * This enumeration defines the argument return values.
+ *
+ */
+enum
+{
+    OT_SIM_OPT_HELP        = 'h',
+    OT_SIM_OPT_SLEEP_TO_TX = 't',
+    OT_SIM_OPT_TIME_SPEED  = 's',
+    OT_SIM_OPT_UNKNOWN     = '?',
+};
+
+static void printUsage(const char *aProgramName, int aExitCode)
+{
+    fprintf(stderr,
+            "Syntax:\n"
+            "    %s [Options] NodeId\n"
+            "Options:\n"
+            "    --help              Display this usage information.\n"
+            "    --sleep-to-tx       Let radio support direct transition from sleep to TX with CSMA.\n"
+            "    --time-speed=val    Speed up the time in simulation.\n",
+            aProgramName);
+
+    exit(aExitCode);
+}
+
 void otSysInit(int aArgCount, char *aArgVector[])
 {
     char *   endptr;
     uint32_t speedUpFactor = 1;
-    int      argIndex      = 0;
+
+    struct option long_options[] = {
+        {"help", no_argument, 0, OT_SIM_OPT_HELP},
+        {"sleep-to-tx", no_argument, 0, OT_SIM_OPT_SLEEP_TO_TX},
+        {"time-speed", required_argument, 0, OT_SIM_OPT_TIME_SPEED},
+        {0, 0, 0, 0},
+    };
 
     if (gPlatformPseudoResetWasRequested)
     {
         gPlatformPseudoResetWasRequested = false;
         return;
     }
 
-    if (aArgCount < 2)
+    while (true)
```

Comment:
`optind` is global variable, initialize it because we use `exec` for resetting. 
```suggestion
    optind = 1;

    while (true)
```